python/cuml/tests/test_umap.py

```diff
@@ -839,6 +839,8 @@ def test_umap_distance_metrics_fit_transform_trust_on_sparse_input(
 def test_umap_trustworthiness_on_batch_nnd(
     num_clusters, fit_then_transform, metric, do_snmg
 ):
+    if do_snmg and num_clusters == 5:
+        pytest.xfail("xfailing snmg test with 5 clusters for now")
```

Comment:
Why is this being xfailed? Can this be removed? If it can't, can you update the reason to be more descriptive of _why_ it's xfailed so future readers can know when it can be removed (and if it's a TODO open a follow-up issue)?

Also: xfail markers like this don't give the test a chance to run and xpass - there's no indicator when running the tests when this can be un-xfailed.

Ideally if a test is flaky or failing we instead handle this by refactoring the parameters to mark the bad case as an xfail. In that way the test still runs, but failures will be ignored.

Reply:
After some investigation it looks like there is an issue in this test when running it on multiple GPUs with parameters (`do_snmg = True` and `num_clusters == 5`). The KNN step returns -1 for the indices. The issue is present in the main branch too. Setting `CUDA_VISIBLE_DEVICES=0` removes the issue explaining why this does not seem to pop up in the CI.

Since, it is not related to this PR I removed the xfail. I will open an issue instead.